app/src/main/java/com/nevilleantony/prototype/adapters/RecyclerViewAdapter.java

```diff
@@ -0,0 +1,81 @@
+package com.nevilleantony.prototype.adapters;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.nevilleantony.prototype.R;
+import com.nevilleantony.prototype.downloadmanager.DownloadRepo;
+import com.nevilleantony.prototype.downloadmanager.FileDownload;
+import com.nevilleantony.prototype.utils.Utils;
+
+import java.util.ArrayList;
+
+public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.RecyclerViewHolder> {
+
+    Context ct;
+    DownloadRepo downloadRepo = DownloadRepo.getInstance(ct);
+    ArrayList downloads = downloadRepo.getDownloads();
+
+    public RecyclerViewAdapter(Context ct) {
+        this.ct = ct;
+    }
+
+    @NonNull
+    @Override
+    public RecyclerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View view = inflater.inflate(R.layout.fragment_recyclerview_row, parent, false);
+        return new RecyclerViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerViewHolder holder, int position) {
+        FileDownload fileDownload = downloadRepo.getFileDownload(downloads.get(position).toString());
+        holder.downloads_name.setText(fileDownload.fileName);
+        holder.downloads_status.setText(R.string.download_status);
+        holder.downloads_size.setText(Utils.getHumanReadableSize(fileDownload.totalFileSize));
+        holder.downloads_percentage.setText("00");
+
+        fileDownload.addOnStateChangedCallback(new FileDownload.OnStateChangedCallback() {
+            @Override
+            public void onStateChanged(FileDownload.DownloadState state) {
+            }
+
+            @Override
+            public void onDownloadComplete() {
+                holder.downloads_percentage.setText("100");
+            }
+
+            @Override
+            public void onProgressChanged(int progress) {
+                String pg = Integer.toString(progress);
+                holder.downloads_percentage.setText(pg);
+            }
+        });
+
```

Comment:
You need to update button's state in onStateChanged, no need to manually set percentage to 100 in onDownloadComplete, and you need to update progress in the progress bar also